### PR TITLE
Reorganize the docker file.

### PR DESCRIPTION
**Pull request by yungyuc**
- Give it a standalone directory contrib/docker/.
- Group related commands together.
- Install miniconda in $HOME/opt/conda3 (which uses Python 3).
- Clean up command remarks.

### CODE REVIEW

**Review by yungyuc — COMMENTED** [review]
@tai271828, I am merging this PR immediately.  Please comment in this PR if you want to discuss my reorganization.

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# This dockerfile builds a docker image containing the latest SOLVCON on git.
+# It be used to quickly set up SOLVCON in docker.
```
Comment (tai271828) [inline contrib/docker/Dockerfile:2]: "It **be** used..." typo?

File: contrib/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ENV SCSRC="$HOME/solvcon"
+# Get latest SOLVCON from github.
+RUN cd $HOME && git clone https://github.com/solvcon/solvcon.git
+# Install most dependencies at user level through conda and pip.
+RUN \
+  mkdir -p $HOME/tmp && \
+  cd $HOME && \
+  wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O /tmp/miniconda3.sh && \
+  bash /tmp/miniconda3.sh -b -p ${CONDA_PREFIX} && \
+  rm -f /tmp/miniconda3.sh
```
Comment (tai271828) [inline contrib/docker/Dockerfile:39]: I like the idea to use /tmp.

**Review by tai271828 — COMMENTED** [review]
The new dockerfile looks good to me. Only one possible typo is in the comments.

File: contrib/docker/Dockerfile (hunk 1 above)
Comment (yungyuc) [reply contrib/docker/Dockerfile:2]: Yes.  Good catch.  I just fixed it.

**Pull request merged**